core/src/main/kotlin/org/opensearch/alerting/core/lock/LockService.kt

```diff
@@ -0,0 +1,311 @@
+                            val parser = XContentType.JSON.xContent()
+                                .createParser(NamedXContentRegistry.EMPTY, LoggingDeprecationHandler.INSTANCE, response.sourceAsString)
+                            parser.nextToken()
+                            listener.onResponse(LockModel.parse(parser, response.seqNo, response.primaryTerm))
+                        } catch (e: IOException) {
+                            log.error("IOException occurred finding lock", e)
+                            listener.onResponse(null)
+                        }
+                    }
+                }
+
+                override fun onFailure(e: Exception) {
+                    log.error("Exception occurred finding lock", e)
+                    listener.onFailure(e)
+                }
+            }
+        )
+    }
+
+    fun release(
+        lock: LockModel?,
+        listener: ActionListener<Boolean>
+    ) {
+        if (lock == null) {
+            log.debug("Lock is null. Nothing to release.")
+            listener.onResponse(false)
+        } else {
+            log.debug("Releasing lock: {}", lock)
+            val lockToRelease = LockModel(lock, true)
+            updateLock(
+                lockToRelease,
+                object : ActionListener<LockModel?> {
+                    override fun onResponse(releasedLock: LockModel?) {
+                        listener.onResponse(releasedLock != null)
+                    }
+
+                    override fun onFailure(e: Exception) {
+                        listener.onFailure(e)
+                    }
+                }
+            )
+        }
+    }
+
+    fun deleteLock(
+        lockId: String,
+        listener: ActionListener<Boolean>
+    ) {
+        val deleteRequest = DeleteRequest(LOCK_INDEX_NAME).id(lockId)
+        client.delete(
+            deleteRequest,
+            object : ActionListener<DeleteResponse> {
+                override fun onResponse(response: DeleteResponse) {
+                    listener.onResponse(
+                        response.result == DocWriteResponse.Result.DELETED || response.result == DocWriteResponse.Result.NOT_FOUND
+                    )
+                }
+
+                override fun onFailure(e: Exception) {
+                    if (e is IndexNotFoundException || e.cause is IndexNotFoundException) {
+                        log.debug("Index is not found to delete lock. {}", e.message)
+                        listener.onResponse(true)
+                    } else {
+                        listener.onFailure(e)
+                    }
+                }
+            }
+        )
+    }
+
+    private fun createLockIndex(listener: ActionListener<Boolean>) {
+        if (lockIndexExist()) {
+            listener.onResponse(true)
+        } else {
+            val indexRequest = CreateIndexRequest(LOCK_INDEX_NAME).mapping(lockMapping())
+                .settings(Settings.builder().put("index.hidden", true).build())
```

Comment:
plz add primary and replica settings based on our general practices for system index.